src/common/datomish/query.cljc

```diff
@@ -63,10 +65,15 @@
   (let [inner-projection (projection/sql-projection-for-relation context)
         inner
         (merge
-          ;; Always SELECT DISTINCT, because Datalog is set-based.
+          ;; Always SELECT DISTINCT, because Datalog is set-based… unless
```

Comment:
Consider putting `Because ..., we always SELECT DISTINCT -- unless...` just to keep the always/unless together.